### PR TITLE
CORE-6335 cleaned up crypto API

### PR DESCRIPTION
**Pull request by alexey-kadyrov-r3**

### CODE REVIEW

**Review by charlieR3 — DISMISSED** [review]

**Review by charlieR3 — DISMISSED** [review]

File: application/src/main/kotlin/net/corda/v5/application/crypto/HashingServiceUtils.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * @param str The algorithm name followed by a delimiter and the sequence of hexadecimal digits that represents a digest.
+ *
+ * @throws IllegalArgumentException The input string does not contain the expected number of hexadecimal digits, or it contains
+ * incorrectly-encoded characters.
+ *
+ * @see SecureHash.create
+ */
+@Suppress("TooGenericExceptionCaught")
+fun HashingService.create(str: String): SecureHash {
```
Comment (mnesbit) [inline application/src/main/kotlin/net/corda/v5/application/crypto/HashingServiceUtils.kt]: I think we generally refer to this as parse on other objects

Comment (alexey-kadyrov-r3) [reply application/src/main/kotlin/net/corda/v5/application/crypto/HashingServiceUtils.kt]: The name was left from C4 but anyway I've renamed it as well as SecureHash.create

**Review by mnesbit — APPROVED** [review]
OK

**Review by ronanbrowne — APPROVED** [review]
LGTM

**Pull request merged**